lib/pdk/validators/base_validator.rb

```diff
@@ -16,7 +20,8 @@ def self.parse_targets(options)
         targets.map { |target|
           if respond_to?(:pattern)
             if File.directory?(target)
-              Array[pattern].flatten.map { |p| Dir.glob(File.join(target, p)) }
+              files_glob = Array[pattern].flatten.map { |p| Dir.glob(File.join(target, p)) }
+              files_glob.flatten.empty? ? target : files_glob
```

Comment:
`puppet-lint` and `puppet parser validate` require a path input to work. The alternative is to not call them at all if no files match.